org.mapstruct.eclipse/src/org/mapstruct/eclipse/internal/MapStructCompletionProposalComputer.java

```diff
@@ -0,0 +1,279 @@
+import org.eclipse.jface.text.contentassist.IContextInformation;
+
+/**
+ * Computes MapStruct specific content assist completion proposals.
+ * 
+ * @author Lars Wetzer
+ */
+public class MapStructCompletionProposalComputer implements IJavaCompletionProposalComputer {
+
+    static final String MAPPING_ANNOTATION_NAME = "Mapping"; //$NON-NLS-1$
+    static final String SOURCE_ANNOTATION_METHOD = "source"; //$NON-NLS-1$
+    static final String TARGET_ANNOTATION_METHOD = "target"; //$NON-NLS-1$
+    static final String GET_PREFIX = "get"; //$NON-NLS-1$
+    static final String SET_PREFIX = "set"; //$NON-NLS-1$
+
+    private static final List<IContextInformation> EMPTY_CONTEXTS = Collections.emptyList();
+    private static final List<ICompletionProposal> EMPTY_PROPOSALS = Collections.emptyList();
+
+    @Override
+    public void sessionStarted() {
+    }
+
+    @Override
+    public void sessionEnded() {
+    }
+
+    @Override
+    public String getErrorMessage() {
+        return null;
+    }
+
+    @Override
+    public List<IContextInformation> computeContextInformation(ContentAssistInvocationContext context,
+                                                               IProgressMonitor monitor) {
+        return EMPTY_CONTEXTS;
+    }
+
+    @Override
+    public List<ICompletionProposal> computeCompletionProposals(ContentAssistInvocationContext context,
+                                                                IProgressMonitor monitor) {
+
+        try {
+
+            if ( !( context instanceof JavaContentAssistInvocationContext ) ) {
+                return EMPTY_PROPOSALS;
+            }
+
+            JavaContentAssistInvocationContext javaContent = (JavaContentAssistInvocationContext) context;
+
+            ICompilationUnit compilationUnit = javaContent.getCompilationUnit();
+
+            if ( compilationUnit == null || !compilationUnit.isStructureKnown() ) {
+                return EMPTY_PROPOSALS;
+            }
+
+            int invocationOffset = javaContent.getInvocationOffset();
+
+            IJavaElement javaElement = compilationUnit.getElementAt( invocationOffset );
+
+            if ( !( javaElement instanceof IMethod ) ) {
+                return EMPTY_PROPOSALS;
+            }
+
+            IMethod method = (IMethod) javaElement;
+
+            for ( IAnnotation annotation : method.getAnnotations() ) {
+
+                if ( MAPPING_ANNOTATION_NAME.equals( annotation.getElementName() )
+                    && isInRange(
+                        invocationOffset,
+                        annotation.getSourceRange().getOffset(),
+                        annotation.getSourceRange().getLength() ) ) {
+
+                    return parseCompilationUnit(
+                        compilationUnit,
+                        invocationOffset,
+                        String.valueOf( javaContent.getCoreContext().getToken() ) );
+
+                }
+
+            }
+
+        }
+        catch ( Exception e ) {
```

Comment:
Content assist is just a fire and forget feature, therefore I think we should not log any errors into the Eclipse log.